bpf/bpf_lxc.c

```diff
@@ -2194,12 +2193,42 @@ int cil_to_container(struct __ctx_buff *ctx)
 #endif
 #ifdef ENABLE_IPV6
 	case bpf_htons(ETH_P_IPV6):
+# ifdef ENABLE_HIGH_SCALE_IPCACHE
+	if (identity == WORLD_ID) {
+		struct endpoint_info *ep;
+		void *data, *data_end;
+		struct ipv6hdr *ip6;
+
+		if (!revalidate_data(ctx, &data, &data_end, &ip6))
+			return DROP_INVALID;
+
+		ep = __lookup_ip6_endpoint((union v6addr *)&ip6->saddr);
```

Comment:
Could be a stupid question: should it be `ip6->daddr`?

If the traffic is from remote, local endpoint map doesn't have an entry for the saddr, right? So this lookup can only be useful for local-to-local traffic. In that case, can we perform policy check at `from-container`'s `handle_ipv4_from_lxc`, where we can get the `dst_sec_id` from local endpoint map if ipcache doesn't return an entry: https://github.com/cilium/cilium/blob/4535f132b70854b9deb9c987cec71a33cff31d00/bpf/bpf_lxc.c#L832-L834